### DISCUSSION TITLE
Different rendering results for GYM envs when using rgb_array rendering mode

### DISCUSSION THREAD

**Discussion by ArgentumWalker** [bulletphysics/bullet3#3374] [General]
Hi! I am trying to get an image as observation from PyBullet Gym Environments and faced that enabling GUI rendering also changes image rendered by `"rgb_array"` mode. For some environments it is crucial.

Code to reproduce:
```
import pybullet_envs
import gym
import matplotlib.pyplot as plt

env = gym.make("HalfCheetahBulletEnv-v0")
env.reset()
plt.imshow(env.render("rgb_array"))
plt.show()

env = gym.make("HalfCheetahBulletEnv-v0")
env.render()
env.reset()
plt.imshow(env.render("rgb_array"))
plt.show()
```

Example images:
![1](https://user-images.githubusercontent.com/17686683/108622171-00833c00-7448-11eb-9ca4-2ba59ec86376.png)
![2](https://user-images.githubusercontent.com/17686683/108622174-06791d00-7448-11eb-8c7a-9b42f7f9213d.png)

**Comment by erwincoumans** [comment]
One uses a simple software CPU renderer (tinyrenderer) and the other a nicer OpenGL3 hardware GPU renderer.

Do you offer to implement the missing features into tinyrenderer?
